docs/scos/dev/upgradability-services/new-documentation/core-method-used-on-project-level.md

```diff
@@ -0,0 +1,88 @@
+---
+title: A core method is used on the project level
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+## A core method is used on the project level
+
+Modules have public and private APIs. Link for API's documentation - [see here]{link to "Private API class was extended or used" - private-api-class-extended-or-used.md#Private API class was extended or used}
+
+{% info_block infoBox "" %}
+
+While public API updates always support backward compatibility, private API updates can break backward compatibility. So, backward compatibility is not guaranteed in the private API. For example, if you use a core method on the project level, and it is updated or removed, it can cause unexpected issues during updates.
+
+{% endinfo_block %}
+
+#### Solution to resolve the issue
+
+To solve this problem with unexpected issues after updating the core's private API, replace the core methods with their copies on the project level.
+Meanwhile, we are working on introducing a way to report such cases and add more extension points in the core.
+
+#### Example of code that causes the upgradability error
+
+CustomerAccessUpdater uses the setContentTypesToInaccessible method from the core level.
+
+```php
+<?php
+namespace Pyz\Zed\CustomerAccess\Business\CustomerAccess;
+...
+class CustomerAccessUpdater extends SprykerCustomerAccessUpdater
+{
+    ...
+    /**
+     * @param \Generated\Shared\Transfer\CustomerAccessTransfer $customerAccessTransfer
```

Comment:
Can we please provide several additional examples, e.g dependency usage, core business model, etc.

Reply:
I think we can, but they will be exactly the same.
Does it make sense? 
Do we have different issues for a DP, business models etc?